### DISCUSSION TITLE
Add example of how to make tsconfig paths work for generate:types

### DISCUSSION THREAD

**Discussion by Stupidism** [payloadcms/payload#1846] [Feature Requests & Ideas]
# Documentation Issue

Today I met an issue that a configured path in my tsconfig.json is not respected by `payload generate:types` command
```
PAYLOAD_CONFIG_PATH=apps/seo-payload/src/payload.config.ts payload generate:types
```

```
Error: Cannot find module '@jerry-serverless/seo-utils/lib/buildPreviewLink'
Require stack:
- /Users/stupidism/Documents/jerry-serverless/apps/seo-payload/src/utils/buildPreviewLink.ts
- /Users/stupidism/Documents/jerry-serverless/apps/seo-payload/src/collections/Articles/Articles.ts
- /Users/stupidism/Documents/jerry-serverless/apps/seo-payload/src/collections/Articles/index.ts
- /Users/stupidism/Documents/jerry-serverless/apps/seo-payload/src/payload.config.ts
- /Users/stupidism/Documents/jerry-serverless/node_modules/payload/dist/config/load.js
- /Users/stupidism/Documents/jerry-serverless/node_modules/payload/dist/bin/generateTypes.js
- /Users/stupidism/Documents/jerry-serverless/node_modules/payload/dist/bin/index.js
- /Users/stupidism/Documents/jerry-serverless/node_modules/payload/bin.js
```

And I couldn't find any help in the [documentation](https://payloadcms.com/docs/typescript/generating-types).

After hours of struggling, I found the workaround with [tsconfig-paths](https://www.npmjs.com/package/tsconfig-paths):

```
PAYLOAD_CONFIG_PATH=apps/seo-payload/src/payload.config.ts ts-node --project apps/seo-payload/tsconfig.app.json -r tsconfig-paths/register ./node_modules/payload/dist/bin/index.js generate:types
```

## Additional Details

I can create a PR to add this advanced usage in the documentation page

**Comment by jmikrut** [comment]
Hey @Stupidism — this would be great to add to the docs. Can you make a PR? This is a new feature / request though rather than a bug so I'll convert this to a discussion and we can keep the conversation going there, but I agree that this would be great to have documented!
